:js
:il
abstract class Effect with
  fun f(): ()

class Effect(f: () -> ())

fun f() =
  handle h = Effect with
    fun f(r) =
      let m = r()
      print("Bye!")
      m
  h.f()
  return 3

// works because return is returning inside the continuation if any effect is invoked
let x = f()
//│ > Bye!
//│ x = 3

fun f() =
  handle h = Effect with
    fun f(r) =
      let m = r()
      print("Bye!")
      m
  in
    h.f()
    return 3
  10

:fixme
:expect 3
let x = f()
//│ > Bye!
//│ ═══[RUNTIME ERROR] Expected: 3, got: 10
